### PR TITLE
Culture innovations/history

### PR DESCRIPTION
**Pull request by Gaben1025**
## How to test:
- Run in-game and tell us your opinion.
- Try and keep an eye out for when the date hits 605.1.1 to see if the AI gets their new techs properly. 
- We have dates for techs before 583, but for now, they are necessary to look at for obvious reasons.

## Changelog:

- Added new history for every race.
     - New dates for when to unlock certain techs.
     - Gave the appropriate amount of techs to cultures based on their tech level in lore.
     - Each culture has a unique history where they will unlock certain techs throughout the new dates.

### CODE REVIEW

File: common/culture/innovations/wc_cultural_innovations.txt [hunk 1]
```diff
@@ -0,0 +1,225 @@
...
+			this = culture:wandering_pandaren
+		  }
+	}
+	modifier = {
+		monthly_lifestyle_xp_gain_mult = 0.5
+	}
+
+	flag = global_regional
+	flag = tribal_era_regular
+}
```
Comment (zumbak04) [inline common/culture/innovations/wc_cultural_innovations.txt:108]: What if the Pandaren follows non-Chi religion like Sargerai or Holy Light?

File: history/cultures/wc_cultures/azerothian.txt [hunk 2]
```diff
@@ -0,0 +1,51 @@
+#Human group
+1.1.1 = {
+	discover_innovation = innovation_motte
+	discover_innovation = innovation_barracks
+	#
+	discover_innovation = innovation_development_01
+}
+
+year.1.1 = {
```
Comment (zumbak04) [inline history/cultures/wc_cultures/azerothian.txt:9]: year?

File: common/culture/innovations/wc_cultural_innovations.txt [hunk 3]
```diff
@@ -0,0 +1,225 @@
...
+	
+		}
+	}
+	modifier = {
+		levy_reinforcement_rate = 0.3
+	}
+
+	flag = global_regional
+	flag = tribal_era_regular
+}
```
Comment (zumbak04) [inline common/culture/innovations/wc_cultural_innovations.txt:134]: What if they follow non-Legion religion?

**Review by zumbak04 — CHANGES_REQUESTED** [review]

File: history/cultures/wc_goblin.txt [hunk 4]
```diff
@@ -0,0 +1,56 @@
+#Human group
+1.1.1 = {
+	discover_innovation = innovation_motte
+	discover_innovation = innovation_barracks
+	#
+	discover_innovation = innovation_development_01
+}
+
+270.1.1.1.1 = {
```
Comment (zumbak04) [inline history/cultures/wc_goblin.txt:9]: Wrong syntax of date.
```suggestion
270.1.1.1.1 = {
```

**Review by zumbak04 — CHANGES_REQUESTED** [review]

File: history/cultures/wc_azerothian.txt [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+}
+
+400.1.1 = {
+	discover_innovation = innovation_bannus
+	#
+	discover_innovation = innovation_gavelkind
+	discover_innovation = innovation_ledger
+	#
+	discover_innovation = innovation_elite_ranks
+	join_era = culture_era_early_medieval
```
Comment (zumbak04) [inline history/cultures/wc_azerothian.txt:25]: It doesn't work. Humans are in the Tribal Era.

**Review by zumbak04 — CHANGES_REQUESTED** [review]

File: history/cultures/wc_azerothian.txt (hunk 5 above)
Comment (zumbak04) [reply history/cultures/wc_azerothian.txt:20]: The bug is still present.
<details>
<summary>Click to expand</summary>

image

</details>

**Review by zumbak04 — CHANGES_REQUESTED** [review]
One bug is unfixed.

File: localization/english/culture/wc_cultures_l_english.yml [hunk 6]
```diff
@@ -200,9 +200,9 @@
  green_dragon_collective_noun:1 "Green Dragons"
  red_dragon:0 "Red Dragon"
  red_dragon_collective_noun:1 "Red Dragons"
- blue_dragon:0 "Blue Dragon"
+ blue_dragon:0 "Blue Dragon"this = culture:vrykul
  blue_dragon_collective_noun:1 "Blue Dragons"
- black_dragon:0 "Black Dragon"
+ blue_dragon:0 "Black Dragon"
```
Comment (zumbak04) [inline localization/english/culture/wc_cultures_l_english.yml:205]: These changes shouldn't be here.

**Review by zumbak04 — CHANGES_REQUESTED** [review]

File: history/cultures/alteraci.txt [hunk 7]
```diff
@@ -0,0 +1,42 @@
+1.1.1 = {
```
Comment (zumbak04) [inline history/cultures/alteraci.txt:1]: This is odd. Cultures don't have innovations that predict other ones.
The Alteraci has the Hereditary Rule, but doesn't have the Gavelkind.
<details>
<summary>Click to expand</summary>

image
image

</details>
The Gilnean has the Bailiffs, but not the Ledger. Why would the Gilnean wait for the early medieval age to get +1 domain limit?
<details>
<summary>Click to expand</summary>

image
image

</details>

**Review by zumbak04 — COMMENTED** [review]
The elves have 19 techs in sum. The amani have 17 techs in sum. I would say... they're too close. It looks like the elves were just a bit better and rushed to the early medieval.
<details>
<summary>Click to expand</summary>

image
image

</details>

**Review by zumbak04 — CHANGES_REQUESTED** [review]
My issue reports aren't addressed.

File: history/cultures/alteraci.txt (hunk 7 above)
Comment (zumbak04) [reply history/cultures/alteraci.txt:1]: It wasn't fixed. All cultures have these problems with different techs. If you used the script, it definitely should be fixed there.
The Gilnean unlocked the Early Medieval military buildings without unlocking the Tribal military buildings.
<details>
<summary>Click to expand</summary>

image
image

</details>
It's like all cultures were rushing to the Early Medieval, spending more progress point while they could just discover the Tribal innovations.
Moreover, it's just not possible in normal games. There's a basic progress increase for every tech. So, if you had time to open 9 early medievals innovations, you already opened all the tribal innovations.

**Review by zumbak04 — CHANGES_REQUESTED** [review]
https://github.com/Warcraft-GoA-Development-Team/Warcraft-Guardians-of-Azeroth-2/pull/41#discussion_r545042453

**Comment by Gaben1025** [comment]
> [#41 (comment)](https://github.com/Warcraft-GoA-Development-Team/Warcraft-Guardians-of-Azeroth-2/pull/41#discussion_r545042453)

**Comment by Gaben1025** [comment]
accdiently closed

**Review by zumbak04 — COMMENTED** [review]
Waiting for changes.

File: common/men_at_arms_types/00_cultural_maa_types.txt [hunk 8]
```diff
@@ -1 +1 @@
-﻿# Warcraft
+﻿# Warcraft
```
Comment (zumbak04) [inline common/men_at_arms_types/00_cultural_maa_types.txt:1]: Just noticed, you should adapt `00_culture_eras.txt` to our timeline.

Comment (Gaben1025) [reply common/men_at_arms_types/00_cultural_maa_types.txt:1]: not really sure how this works, different cultures will unlock eras at different times, I updated it but I'm not sure this is what you're looking for.

Comment (zumbak04) [reply common/men_at_arms_types/00_cultural_maa_types.txt:1]: > not really sure how this works, different cultures will unlock eras at different times, I updated it but I'm not sure this is what you're looking for.

Oops, I haven't noticed you were already adapted.

Though `culture_era_late_medieval` is too close to `culture_era_high_medieval`. In vanilla, there's 150 years gap.

File: history/cultures/alteraci.txt [hunk 9]
```diff
@@ -0,0 +1,51 @@
...
+}
+400.1.1 = {
+	discover_innovation = innovation_quilted_armor 
+	discover_innovation = innovation_motte 
+	#
+	discover_innovation = innovation_currency_01 
+	discover_innovation = innovation_casus_belli 
+	discover_innovation = innovation_plenary_assemblies 
+	#
+	discover_innovation = innovation_elite_ranks
```
Comment (zumbak04) [inline history/cultures/alteraci.txt:26]: Huh? Remember, I moved new cultural innovations to a separate branch.

File: history/cultures/alteraci.txt [hunk 10]
```diff
@@ -0,0 +1,51 @@
+1.1.1 = {
```
Comment (zumbak04) [inline history/cultures/alteraci.txt:1]: Night elves have to techs.
<details>
<summary>Click to expand</summary>

image

</details>

**Review by zumbak04 — CHANGES_REQUESTED** [review]

File: history/cultures/night_elfdryad.txt [hunk 11]
```diff
@@ -0,0 +1,49 @@
+1.1.1 = {
```
Comment (zumbak04) [inline history/cultures/night_elfdryad.txt:1]: HMM! Two races! One culture!

**Review by zumbak04 — CHANGES_REQUESTED** [review]

**Review by zumbak04 — APPROVED** [review]
LGTM!

**Comment by zumbak04** [comment]
Correct the changelog.

> Added new history for every race.

History of what?

**Review by TheGroxEmpire — COMMENTED** [review]
- It's weird to have Dalaranian culture less advanced than Gilnean, Kultiran, and Tol Baradin. They are too far advanced, on the same level as Quel'dorei innovations. It should be Dalaranian that is on that level.
- Passing 605.1.1 doesn't unlock the supposed innovations for AI

File: history/cultures/lordaeronian.txt [hunk 12]
```diff
@@ -0,0 +1,49 @@
...
+583.1.1 = {
+	discover_innovation  = innovation_horseshoes 
+	discover_innovation  = innovation_mangonel 
+	#
+	discover_innovation = innovation_manorialism 
+	discover_innovation = innovation_currency_02 
+}
+605.1.1 = {
+	discover_innovation  = innovation_burhs 
+	discover_innovation  = innovation_arched_saddle
```
Comment (TheGroxEmpire) [inline history/cultures/lordaeronian.txt:44]: - We know for a fact that armored horses existed on Warcraft II (https://wow.gamepedia.com/Knight_(Warcraft_II)). Eastern Kigndom human group (Lordaeronian, Azerothian, Alteraci, Hilsbradian, and Stromic) should have at least ``discover_innovation  = innovation_arched_saddle`` after 587.1.1 (https://wow.gamepedia.com/Timeline#Second_War)

**Review by TheGroxEmpire — COMMENTED** [review]
- We know for a fact that armored horses existed on Warcraft II (https://wow.gamepedia.com/Knight_(Warcraft_II)). Eastern Kigndom human group (Lordaeronian, Azerothian, Alteraci, Hilsbradian, and Stromic) should have at least ``discover_innovation  = innovation_arched_saddle`` after 587.1.1 (https://wow.gamepedia.com/Timeline#Second_War)

**Comment by zumbak04** [comment]
> * Passing 605.1.1 doesn't unlock the supposed innovations for AI

Can't confirm. Techs are more advanced in 605 for me.
<details>
<summary>583 Bookmark</summary>

image

</details>
<details>
<summary>605 Bookmark</summary>

image

</details>

**Review by zumbak04 — APPROVED** [review]
LGTM!

**Comment by TheGroxEmpire** [comment]
> > * Passing 605.1.1 doesn't unlock the supposed innovations for AI
> 
> Can't confirm. Techs are more advanced in 605 for me.
> 
> 583 Bookmark
> 605 Bookmark

Hmm. I misinterpreted the test checklist then. From the wording of it, I thought it should pass from in-game time too, not just bookmark.

**Pull request merged**